p2p/transport/webrtc/udp_mux.go

```diff
@@ -0,0 +1,341 @@
+package libp2pwebrtc
+
+import (
+	"errors"
+	"io"
+	"net"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/pion/ice/v2"
+	"github.com/pion/logging"
+	"github.com/pion/stun"
+)
+
+type candidateAddr struct {
+	raddr *net.UDPAddr
+	ufrag string
+}
+
+// UDPMuxNewAddr is an implementation of the interface
+type UDPMuxNewAddr struct {
```

Comment:
1. Why is this exposed?
2. I don't understand what this struct does. The comment doesn't say anything. This will need a multi-line comment with a detailed explanation.
3. Add a compile time assertion for the implemented interface (see other comments).